### ISSUE TITLE
Passing in loader data?

### ISSUE THREAD

**Issue by soluml** [ampedandwired/html-webpack-plugin#77]
So, I'm fairly new to Webpack, so hopefully this is a good place to ask this.

This request is sort of related to #10, but instead of passing in loader data directly I was hoping to pass it in as a property that can be accessed from the template.  I was hoping to pass in my localized styles from css-loader.  Something like this:

**webpack.config**:

    new HtmlWebpackPlugin({
        styles: require('style-loader!css-loader?localIdentName=[local]__[hash:base64:5]!autoprefixer-loader?{browsers:["last 2 version", "ie 9"]}!sass-loader!src/scss/main/s.scss'),
        filename: 'main.handlebars',
        template: 'src/tmpl/main.html'
    })

**main.html**:

    <!DOCTYPE html>
    <html>
        <head>
            <meta http-equiv="Content-type" content="text/html; charset=utf-8"/>
            <title>Title</title>
        </head>
        <body class="{%= o.htmlWebpackPlugin.options.styles.bodyClass %}">
        </body>
    </html>

Unfortunately, I can't seem to find a configuration that allows me to process a loader in the config.  Is this possible?

Thanks!

**Comment by jantimon** [comment]
You should set `inject:true`and move the require to your js file

**Comment by soluml** [comment]
Thanks for the quick reply. Unless I'm missing something, seems like `inject: true` just sticks the style/script tags into the template, but that's not what I'm after.

When I require a CSS file in a JS file `var styles = require('../../scss/main/s.scss');`, I'm given an object keyed by the original class names with the obfuscated names as the values:

    {
       "bodyClass": "bodyClass__a7asj"
     }

I've already required the css file in my JS, so I can use it in all my JS stuff, but I can't seem to find a way to pass this object to my templates.

**Comment by jantimon** [comment]
This isn't possible right now - you have to generate the html using a javascript template language for that

**Comment by jantimon** [comment]
e.g. EJS or React

**Comment by soluml** [comment]
Thanks, I was afraid of that.  I was trying to take these templates than pass them over to Node.js for rendering on the server with React.  Guess that's not going to work :/

**Comment by jantimon** [comment]
You still can render webpack content on the server.

**Comment by soluml** [comment]
I tried something like this in Node running Express: `var styles = require('css-loader?localIdentName=[local]__[hash:base64:5]!autoprefixer-loader?{browsers:["last 2 version", "ie 9"]}!sass-loader!src/scss/main/s.scss')`

All I got was errors and all the loaders were definitely present in my package.json file as I'm able to use them in webpack for my front-end JS.  Am I completely missing something?  I've poured over Webpack docs and Stackoverflow and I'm just not finding many examples.
